task_02/alexey_mesheryakov/graph.cpp

```diff
@@ -0,0 +1,91 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <utility>
+
+const int LEN = 14;
+
+using std::pair;
+using std::vector;
+using std::cout;
+
+class Graph {
+public:
+	struct Edge{
+		int self_id = 0;
+		pair <int, int> vert = {-1,-1};
+	};
+	
+	struct Vert{
+		int inf;
+		vector<int> edges = {};
+	};
+
+	void set_vert_inf(int id, int inf) {
+		vert_inf[id].inf = inf;
+	}
+	
+	void set_vert(Vert v) {
+		vert_inf.push_back(v);  
+	}
+
+	void set_vert_inf(vector<Vert> inf) {
+		vert_inf = inf;
+	}
+
+	void set_edge(Edge edge) {
+	 	edges.push_back(edge);
+		vert_inf[edge.vert.first].edges.push_back(edge.self_id);
+		vert_inf[edge.vert.second].edges.push_back(edge.self_id);
+	}
+
+	Graph (vector<Edge> newedges, vector<int> inf) {
```

Comment:
Обычно вибирают один из 2ух популярных способов именования переменных `snake_case` или `camelCase`. Учитывая, что до этого ты придерживался `snake_case`, то предлагаю так и продолжать
```suggestion
      Graph (vector<Edge> new_edges, vector<int> inf) {
```

Reply:
Так как вектор, передаваемый в конструктор, может быть довольно большим, то было бы неплохо попытаться избежать копирования. Обычно в таких случаях аргумент передают по `const` ссылке.
```suggestion
      Graph (const vector<Edge>& newedges, const vector<int>& inf) {
```

Reply:
Зачем ты усложняешь конструктор. Граф будет валидным объектом даже без вершин и граней. Совершенно не обязательно на этапе конструкции объекта уже их добавлять.
И тем более, это не совсем ответственность графа - конструировать самого себя из вершин и граней. Если действительно, есть такая необходимость, то, скорее для этого должна использоваться некая фабрика или генератор.
```suggestion
      Graph() {
```